runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -0,0 +1,217 @@
+import org.corfudb.runtime.object.transactions.TransactionType;
+
+import org.corfudb.runtime.object.transactions.TransactionalContext;
+import org.corfudb.util.serializer.ISerializer;
+import lombok.Getter;
+
+/**
+ * Created by zlokhandwala on 2019-08-05.
+ */
+public class Table<K extends Message, V extends Message, M extends Message> {
+
+    private final CorfuRuntime corfuRuntime;
+
+    private final CorfuTable<K, CorfuRecord<V, M>> corfuTable;
+
+    /**
+     * Namespace this table belongs in.
+     */
+    @Getter
+    private final String namespace;
+
+    /**
+     * Fully qualified table name: created by the namespace and the tablename.
+     */
+    @Getter
+    private final String fullyQualifiedTableName;
+
+    @Getter
+    private final MetadataOptions metadataOptions;
+
+    /**
+     * Returns a Table instance backed by a CorfuTable.
+     *
+     * @param namespace               Namespace of the table.
+     * @param fullyQualifiedTableName Fully qualified table name.
+     * @param valueSchema             Value schema to identify secondary keys.
+     * @param corfuRuntime            Connected instance of the Corfu Runtime.
+     * @param serializer              Protobuf Serializer.
+     */
+    @Nonnull
+    public Table(@Nonnull final String namespace,
+                 @Nonnull final String fullyQualifiedTableName,
+                 @Nonnull final V valueSchema,
+                 @Nullable final M metadataSchema,
+                 @Nonnull final CorfuRuntime corfuRuntime,
+                 @Nonnull final ISerializer serializer) {
+
+        this.corfuRuntime = corfuRuntime;
+        this.namespace = namespace;
+        this.fullyQualifiedTableName = fullyQualifiedTableName;
+        if (metadataSchema != null) {
+            this.metadataOptions = MetadataOptions.builder()
+                    .metadataEnabled(true)
+                    .defaultMetadataInstance(metadataSchema)
+                    .build();
+        } else {
+            this.metadataOptions = MetadataOptions.<M>builder().build();
+        }
+        this.corfuTable = corfuRuntime.getObjectsView().build()
+                .setTypeToken(new TypeToken<CorfuTable<K, CorfuRecord<V, M>>>() {
+                })
+                .setStreamName(this.fullyQualifiedTableName)
+                .setSerializer(serializer)
+                .setArguments(new ProtobufIndexer(valueSchema))
+                .open();
+    }
+
+    /**
+     * Begins a write after write transaction.
+     */
+    private boolean TxBegin() {
+        if (!TransactionalContext.isInTransaction()) {
+            corfuRuntime.getObjectsView()
+                    .TXBuild()
+                    .type(TransactionType.OPTIMISTIC)
+                    .build().begin();
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Ends an ongoing transaction.
+     */
+    private void TxEnd() {
+        corfuRuntime.getObjectsView().TXEnd();
```

Comment:
how table can manage transactions?
It violates logic of transactions

Reply:
You should have some sort of a TxManager for this logic.